apis/python/src/tiledbsoma/io/ingest.py

```diff
@@ -214,6 +216,10 @@ class _FSPathWrapper:
     so here we just proxy all attributes except ``__fspath__``.
     """
 
+    # Needed to make typeguard happy
+    def __call__(self) -> None:
+        pass
```

Comment:
might be worth making this `raise AssertionError("this should never be called")` since, well, it shouldn't

Reply:
I actually removed this altogether, it didn't seem to be needed.

#2291 also includes https://github.com/single-cell-data/TileDB-SOMA/pull/2291/commits/2f00a9b3f7dd81910b43fec856d101e456f44f6d, which makes `_FSPathWrapper` inherit from `pathlib.Path`, which mitigates further type-checking issues surrounding `_FSPathWrapper` (and seems like a good change to make anyway).